browser_use/controller/service.py

```diff
@@ -384,6 +386,60 @@ async def scroll_down(params: ScrollAction, browser: BrowserContext):
 				include_in_memory=True,
 			)
 
+		class ImageDownloadParams(BaseModel):
+			urls: list[str]
+			save_path: str | None = None
+
+		@self.registry.action(
+			'Download and save a list of image URLs to local files',
+			param_model=ImageDownloadParams,
+		)
+		async def download_images(params: ImageDownloadParams, browser: BrowserContext) -> ActionResult:
+			"""Downloads a list of image URLs asynchronously and saves them as files."""
+			try:
+				save_path = params.save_path or os.getcwd()
+				os.makedirs(save_path, exist_ok=True)
+
+				results = []
+				async with aiohttp.ClientSession() as session:  # Create an async HTTP session
+					tasks = []
+					for url in params.urls:
+						tasks.append(fetch_and_save(session, url, save_path, results))
+
+					# Run all downloads concurrently
+					await asyncio.gather(*tasks)
```

Comment:
Some websites might may concurrent download of many assets. 

We should be able to enable/disable concurrent downloading of images with a parameter:

```
concurrency
```

* 0 means enabled
* 1 means disabled
* Any number > 1, will be the desired level of concurrency (e.g. 4)

Or using union types:

```
concurrency: "enabled" | "disabled" | Int = "disabled"
```

Reply:
I'd be surprised if that is a problem in practice, most browsers send 3 or more requests in parallel per host during normal pageloads. If we actually see reports of this being a problem we can add it, but I dont want to add more complex concurrency control unless it's absolutely needed.